main.js

```diff
@@ -65,6 +88,63 @@ function displayMenu() {
     document.getElementById("timer").style.visibility = "visible";
 }
 
+// calculate the numbers of bombs around each cell
+function getDistanceBomb(n) {
+    for (let i = 0; i < n; ++i) {
+        for (let j = 1; j < n; ++j) {
+            if (board[i][j] != "bomb") {
+                if (i > 0 && board[i - 1][j] == "bomb") {
+                    ++board[i][j];
+                }
+                if (i < n - 1 && board[i + 1][j] == "bomb") {
+                    ++board[i][j];
+                }
+                if (j > 0 && board[i][j - 1] == "bomb") {
+                    ++board[i][j];
+                }
+                if (j < n - 1 && board[i][j + 1] == "bomb") {
+                    ++board[i][j];
+                }
+                if (i > 0 && j < n - 1 && board[i - 1][j + 1] == "bomb") {
+                    ++board[i][j];
+                }
+                if (j > 0 && i < n - 1 && board[i + 1][j + 1] == "bomb") {
+                    ++board[i][j];
+                }
+                if (j > 0 && i < n - 1 && board[i + 1][j - 1] == "bomb") {
+                    ++board[i][j];
+                }
+                if (i > 0 && j < n - 1 && board[i - 1][j - 1] == "bomb") {
+                    ++board[i][j];
+                }
+            } 
+        }
+    }
+}
+
+// plants the bomns in the grid in random positions
+function plantBombs(noBombs) {
+    for (let i = 0; i < noBombs; ++i) {
+        board[randomCell(noBombs - 1)][randomCell(noBombs - 1)] = "bomb";
+    }
+}
+
+// returns a random value within a range
+function randomCell(max) {
+    const floatRandom = Math.random();
+    const random = Math.round(max * floatRandom);
+    return random;
+}
+
+// overwrite the matrix in the table
+function refactorId(sizeMatrix) {
+    for (let i = 0, cellIndex = 0; i < sizeMatrix; ++i) {
+        for (let j = 0; j < sizeMatrix; ++j, ++cellIndex) {
+            unit[cellIndex].id = board[i][j];
```

Comment:
You need to find a better way of marking the bombs, because you don't want the user to see where the bombs are